File: src/main/java/sprykerFileCreator/model/definition/spryker/ZedCommunicationFactoryDefinition.java
package sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ZedCommunicationFactoryDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.ZED_COMMUNICATION_FACTORY;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}CommunicationFactory";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Zed\\{{ bundleName }}\\Communication";
    }

    @Override
    public String getMethodForReturnType() {
        return "getFactory()";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {
                SprykerConstants.ZED_QUERY_CONTAINER,
                SprykerConstants.ZED_FACADE,
                SprykerConstants.ZED_CONFIG
        };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Zed\\Kernel\\Communication\\AbstractCommunicationFactory";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/services/definitions/spryker/ZedCommunicationFactoryDefinition.kt
package com.github.tobi812.sprykerplugin.services.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.services.definitions.ClassDefinitionInterface

class ZedCommunicationFactoryDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.ZED_COMMUNICATION_FACTORY
    override val namePattern: String
        get() = "{{ bundleName }}CommunicationFactory"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Zed\\{{ bundleName }}\\Communication"
    override val methodForReturnType: String
        get() = "getFactory()"
    override val docBlockClasses: Array<String>
        get() = arrayOf(
                SprykerConstants.ZED_QUERY_CONTAINER,
                SprykerConstants.ZED_CONFIG
        )
    override val defaultParentClass: String
        get() = "Spryker\\Zed\\Kernel\\Communication\\AbstractCommunicationFactory"
}
